File: software_challenge_sdk/src/server_api/sc/protocol/responses/RoomPacket.java
package sc.protocol.responses;

import com.thoughtworks.xstream.annotations.XStreamAlias;
import com.thoughtworks.xstream.annotations.XStreamAsAttribute;

import sc.protocol.requests.ILobbyRequest;

@XStreamAlias("room")
public final class RoomPacket implements ILobbyRequest
{
	@XStreamAsAttribute
	private String	roomId;
	
	private Object	data;

        /**
         * might be needed by XStream
         */
        public RoomPacket() {
        }

	public RoomPacket(String roomId, Object o)
	{
		this.roomId = roomId;
		this.data = o;
	}

	public String getRoomId()
	{
		return this.roomId;
	}

	public Object getData()
	{
		return this.data;
	}
}

File: sdk/src/server-api/sc/protocol/RoomPacket.kt
package sc.protocol

import com.thoughtworks.xstream.annotations.XStreamAlias
import com.thoughtworks.xstream.annotations.XStreamAsAttribute
import sc.protocol.RoomMessage

import sc.protocol.requests.ILobbyRequest

/** Used to send a [RoomMessage] in a room as an [ILobbyRequest]. */
@XStreamAlias("room")
data class RoomPacket(
        @XStreamAsAttribute
        val roomId: String,
        val data: RoomMessage
): ILobbyRequest
